orca-queue-redis/src/main/kotlin/com/netflix/spinnaker/orca/q/redis/RedisActiveExecutionsMonitor.kt

```diff
@@ -0,0 +1,172 @@
+          .filter { it.id() == expectedId }
+          .findFirst()
+          .orElse(registry.gauge(expectedId, null, {
+            snapshot[expectedId]?.toDouble() ?: 0.0
+          }))
+      }
+    }
+  }
+
+  private fun snapshotActivity(): List<ActiveExecution> {
+    log.info("Snapshotting active executions")
+
+    val activeExecutions = getActiveExecutions()
+
+    val working = mutableMapOf<Id, AtomicLong>()
+    activeExecutions.map { it.getMetricId() }.forEach {
+      if (working[it] == null) {
+        working[it] = AtomicLong()
+      }
+      working[it]?.incrementAndGet()
+    }
+
+    // Update snapshot from working copy
+    working.forEach { snapshot[it.key] = it.value }
+
+    // Remove keys that are not in the working copy
+    snapshot.keys
+      .filterNot { working.containsKey(it) }
+      .forEach { snapshot.remove(it) }
+
+    return activeExecutions
+  }
+
+  @Scheduled(fixedDelayString = "\${queue.monitor.activeExecutions.cleanup.frequency.ms:300000}")
+  fun cleanup() {
+    val orphans = getActiveExecutions()
+      .map {
+        val execution: Execution
+        try {
+          execution = executionRepository.retrieve(it.type, it.id)
+        } catch (e: ExecutionNotFoundException) {
+          return@map it.id
+        }
+        return@map if (execution.status.isComplete) it.id else null
+      }
+      .filterNotNull()
+      .toTypedArray()
+
+    log.info("Cleaning up ${orphans.size} orphaned active executions")
+    if (orphans.isNotEmpty()) {
+      pool.resource.use { redis ->
+        redis.hdel(REDIS_KEY, *orphans)
+      }
+    }
+  }
+
+  override fun onApplicationEvent(event: ExecutionEvent) {
+    if (event is ExecutionStarted) {
+      startExecution(event.executionType, event.executionId)
+    } else if (event is ExecutionComplete) {
+      completeExecution(event.executionId)
+    }
+  }
+
+  private fun startExecution(executionType: Execution.ExecutionType, executionId: String) {
+    val execution: Execution
+    try {
+      execution = executionRepository.retrieve(executionType, executionId)
+    } catch (e: ExecutionNotFoundException) {
+      log.error("Received start execution event, but was unable to read execution from the database")
+      return
+    }
+
+    pool.resource.use { redis ->
+      redis.hset(REDIS_KEY, execution.id, objectMapper.writeValueAsString(ActiveExecution(
+        id = execution.id,
+        type = execution.type,
+        application = execution.application
+      )))
+    }
+  }
+
+  private fun completeExecution(executionId: String) {
+    pool.resource.use { redis ->
+      redis.hdel(REDIS_KEY, executionId)
+    }
+  }
+
+  private fun getActiveExecutions() =
+    pool.resource.use { redis ->
+      redis.hgetAll(REDIS_KEY).map { objectMapper.readValue(it.value, ActiveExecution::class.java) }
+    }
+
+  private fun ActiveExecution.getMetricId() =
+    registry.createId("executions.active")
+      .withTag("type", type.toString())
```

Comment:
Maybe call this `executionType` to be consistent with other metrics (`task.invocations` at least)